src/Xamarin.Android.Build.Tasks/Xamarin.Android.Build.Tasks.targets

```diff
@@ -120,6 +123,56 @@
     <_SharedRuntimeAssemblies Include="$(_SharedRuntimeBuildPath)$(AndroidFrameworkVersion)\OpenTK.dll" />
     <_SharedRuntimeAssemblies Include="$(_SharedRuntimeBuildPath)$(AndroidFrameworkVersion)\OpenTK-1.0.dll" />
   </ItemGroup>
+  <Target Name="_PrepareNDKToolsItems">
+    <PropertyGroup>
+      <_PrebuiltToolchainDir Condition=" '$(HostOS)' == 'Linux' ">linux-x86_64</_PrebuiltToolchainDir>
+      <_PrebuiltToolchainDir Condition=" '$(HostOS)' == 'Darwin' ">darwin-x86_64</_PrebuiltToolchainDir>
+      <_PrebuiltToolchainDir Condition=" '$(HostOS)' == 'Windows' And '$(HostBits)' == '64' ">windows-x86_64</_PrebuiltToolchainDir>
+      <_PrebuiltToolchainDir Condition=" '$(HostOS)' == 'Windows' And '$(HostBits)' == '32' ">windows</_PrebuiltToolchainDir>
+    </PropertyGroup>
+    <ItemGroup>
+      <_ToolchainPrefix Include="aarch64-linux-android">
+        <ToolPrefix>aarch64-linux-android</ToolPrefix>
+      </_ToolchainPrefix>
+      <_ToolchainPrefix Include="arm-linux-androideabi">
+        <ToolPrefix>arm-linux-androideabi</ToolPrefix>
+      </_ToolchainPrefix>
+      <_ToolchainPrefix Include="x86">
+        <ToolPrefix>i686-linux-android</ToolPrefix>
+      </_ToolchainPrefix>
+      <_ToolchainPrefix Include="x86_64">
+        <ToolPrefix>x86_64-linux-android</ToolPrefix>
+      </_ToolchainPrefix>
+    </ItemGroup>
+
+    <Which
+        HostOS="$(HostOS)"
+        HostOSName="$(HostOsName)"
+        Program="$(AndroidNdkFullPath)/toolchains/%(_ToolchainPrefix.Identity)-4.9/prebuilt/$(_PrebuiltToolchainDir)/bin/%(_ToolchainPrefix.ToolPrefix)-as"
```

Comment:
I assume this (and related) logic is  what copies the `-as` tool from the NDK install into the output directory .

Please update `Xamarin.Android.Build.Tasks.tpnitems` so that this is noted within `ThirdPartyNotices.txt`:

```diff
diff --git a/src/Xamarin.Android.Build.Tasks.tpnitems b/src/Xamarin.Android.Build.Tasks.tpnitems
index 4de6e1e5..0f302236 100644
--- a/src/Xamarin.Android.Build.Tasks.tpnitems
+++ b/src/Xamarin.Android.Build.Tasks.tpnitems
@@ -122,4 +122,25 @@
       <SourceUrl>https://github.com/NuGet/NuGet.Client</SourceUrl>
     </ThirdPartyNotice>
   </ItemGroup>
+  <!-- `Xamarin.Android.Build.Tasks/Xamarin.Android.Build.Tasks.csproj` copies various NDK utilities into the SDK -->
+  <ItemGroup Condition=" '$(TpnIncludeExternalDependencies)' == 'True' ">
+    <ThirdPartyNotice Include="android/platform/ndk">
+      <LicenseText>
+      Copyright (C) 2016 The Android Open Source Project
+
+      Licensed under the Apache License, Version 2.0 (the "License");
+      you may not use this file except in compliance with the License.
+      You may obtain a copy of the License at
+
+           http://www.apache.org/licenses/LICENSE-2.0
+
+      Unless required by applicable law or agreed to in writing, software
+      distributed under the License is distributed on an "AS IS" BASIS,
+      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+      See the License for the specific language governing permissions and
+      limitations under the License.
+      </LicenseText>
+      <SourceUrl>https://android.googlesource.com/platform/ndk/+/master</SourceUrl>
+    </ThirdPartyNotice>
+  </ItemGroup>
 </Project>
```

Additionally, we need to double (triple?) check that the utilities within the NDK are *actually* available under Apache 2, and not just the source code.  Rephrased: do we need to build the NDK *ourselves* to redistribute these utilities?

Reply:
`as` is distributed under GPL as it's part of GNU Binutils which gives us the right to redistribute. The binaries shipped by NDK [are not built](https://android.googlesource.com/platform/ndk/+/master/docs/Toolchains.md) as part of the NDK build itself, but rather prebuilt and then packaged with the NDK. I updated the license accordingly.

Reply:
It gives us the right to redistribute so long as we follow the GPL licensing terms.

If we don't know what source was used to build `as`, it becomes *very difficult* to actually follow those licensing terms, as we need to be able to provide the source used to compile it, and *we don't know what source was used to compile it*.

This sounds problematic. :-)

It would probably be safer/saner to actually build `binutils` ourselves.  At least then we'd know what we're redistributing.

Reply:
Turns out binutils are dual-licensed - GPL2 or GPL3, and `gas` in particular is `GPL3`